app/main.py

```diff
@@ -1,34 +1,113 @@
+from typing import List
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    counter = 0
+
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
+
+    def __repr__(self) -> str:
+        return f"Deck: {self.row} : {self.column}"
+
+    def __str__(self) -> str:
+        return f"Deck: {self.row} : {self.column}"
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def __init__(self, start: tuple, end: tuple,
+                 is_drowned: bool = False) -> None:
+        self.decks = self._decks_generator(start, end)
+        self.is_drowned = is_drowned
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+
+    def fire(self, row: int, column: int) -> bool:
+        deck = self.get_deck(row, column)
+        if deck and deck.is_alive:
+            deck.is_alive = False
+        decks_available = [deck for deck in self.decks if deck.is_alive]
+        if not decks_available:
+            self.is_drowned = True
+
+    @staticmethod
+    def _decks_generator(start: tuple, end: tuple) -> list:
+        start_row, start_column = start
+        end_row, end_column = end
+        if start_row == end_row:
+            return [Deck(start_row, column)
+                    for column in range(start_column, end_column + 1)]
+        return [Deck(row, start_column)
+                for row in range(start_row, end_row + 1)]
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+
+    def __init__(self, ships: List[Ship]) -> None:
+        self.field = self.get_field(ships)
+        self.draw_playfield()
+
+    def _validate_fields(self) -> str:
+        ships = self.field.values()
+        if set(ships) != 10:
```

Comment:
I added this check since your validation method should check if number of shits is exactly 10 and since multiple key: value pairs may have reference to same ship I had to make set of it